File: Problems/Algorithms/338. Counting Bits/CountingBits.kt
class Solution {
    fun countBits(n: Int): IntArray {
        return IntArray(n+1) { idx -> getBit(idx) }
    }
    
    private fun getBit(n: Int): Int {
        var count: Int = 0
        var num = n
        while (num > 0) {
            count++
            num = num and (num-1)
        }
        return count
    }
}

File: Problems/Algorithms/338. Counting Bits/counting_bits.py
#!/usr/bin/env python3

from typing import List


class Solution:
    def countBits(self, n: int) -> List[int]:
        curr = 0
        ans = []

        for i in range(n+1):
            curr = bin(i).count("1")
            ans.append(curr)

        return ans
